pulser-core/pulser/sequence/_seq_drawer.py

```diff
@@ -670,6 +534,306 @@ def phase_str(phi: float) -> str:
+    area_ph_box = dict(boxstyle="round", facecolor="ghostwhite", alpha=0.7)
+    ph_box = dict(boxstyle="round", facecolor="ghostwhite")
+
+    (fig_reg, fig, ch_axes, data) = draw_channel_content(
+        data,
+        sampled_seq,
+        seq.register if draw_register else None,
+        sampling_rate,
+        draw_input,
+        draw_modulation,
+        draw_phase_curve,
+    )
+
+    total_duration = data["total_duration"]
+    time_scale = 1e3 if total_duration > 1e4 else 1
+    t = np.arange(total_duration) / time_scale
+    final_t = t[-1]
+    t_min = -final_t * 0.03
+    t_max = final_t * 1.05
+
+    for ch, axes in ch_axes.items():
+        ch_obj = seq.declared_channels[ch]
+        ch_data = data[ch]
+        basis = ch_obj.basis
+        ys = ch_data.get_input_curves()
+        ys_mod = [()] * 3
+        yseff = [()] * 3
+        draw_output = draw_modulation and (
+            ch_obj.mod_bandwidth or not draw_input
+        )
+        if draw_output:
+            ys_mod = ch_data.get_output_curves(ch_obj)
+
+        if sampling_rate:
+            curves = ys_mod if draw_output else ys
+            yseff = ch_data.interpolate_curves(curves, sampling_rate)
+        ref_ys = [
+            list(chain.from_iterable(all_ys))
+            for all_ys in zip(ys, ys_mod, yseff)
+        ]
+        max_amp = np.max(ref_ys[0])
+        max_amp = 1 if max_amp == 0 else max_amp
+        amp_top = max_amp * 1.2
+        amp_bottom = min(0.0, *ref_ys[0])
+        # Makes sure that [-1, 1] range is always represented
+        det_max = max(*ref_ys[1], 1)
+        det_min = min(*ref_ys[1], -1)
+        det_range = det_max - det_min
+        det_top = det_max + det_range * 0.15
+        det_bottom = det_min - det_range * 0.05
+        ax_lims = [
+            (amp_bottom, amp_top),
+            (det_bottom, det_top),
+            (min(0.0, *ref_ys[2]), max(1.1, *ref_ys[2])),
+        ]
+        ax_lims = [ax_lims[i] for i in ch_data.curves_on_indices()]
+
+        if draw_phase_area:
+            top = False  # Variable to track position of box, top or center.
+            print_phase = not draw_phase_curve and any(
+                seq_.type.phase != 0
+                for seq_ in seq._schedule[ch]
+                if isinstance(seq_.type, Pulse)
+            )
+            for pulse_num, seq_ in enumerate(seq._schedule[ch]):
+                # Select only `Pulse` objects
+                if isinstance(seq_.type, Pulse):
+                    if sampling_rate:
+                        area_val = (
+                            np.sum(yseff[0][seq_.ti : seq_.tf]) * 1e-3 / np.pi
+                        )
+                    else:
+                        area_val = seq_.type.amplitude.integral / np.pi
+                    phase_val = seq_.type.phase
+                    x_plot = (seq_.ti + seq_.tf) / 2 / time_scale
+                    if (
+                        seq._schedule[ch][pulse_num - 1].type == "target"
```

Comment:
[draw_phase_area] Replace by `slot.ti in [target_slot.tf for target_slot in sampled_seq.channel_samples[ch].time_slots_target]`